docs/code-search/code-navigation/auto_indexing.mdx

```diff
@@ -0,0 +1,320 @@
+
+- **`PRECISE_CODE_INTEL_AUTO_INDEXING_TASK_INTERVAL`**: The frequency with which to run periodic codeintel auto-indexing tasks. Default is every 2 minutes
+- **`PRECISE_CODE_INTEL_AUTO_INDEXING_REPOSITORY_PROCESS_DELAY`**: The minimum frequency that the same repository can be considered for auto-index scheduling. Default is every 24 hours
+- **`PRECISE_CODE_INTEL_AUTO_INDEXING_REPOSITORY_BATCH_SIZE`**: The number of repositories to consider for auto-indexing scheduling at a time. Default is 2500
+- **`PRECISE_CODE_INTEL_AUTO_INDEX_MAXIMUM_REPOSITORIES_INSPECTED_PER_SECOND`**: The maximum number of repositories inspected for auto-indexing per second. Set to zero to disable limit. Default is 0
+
+### Access to private repositories and packages
+
+Auto-indexing jobs run as Docker containers on the Executors infrastructure, and by default don't have any knowledge about the private repositories or package registries your builds might use.
+
+The type of information you will need to provide depends on the language and ecosystem, but the mechanism is the same - [adding a secret to Executors "Code Graph" namespace](/admin/executors/executor_secrets).
+
+See the [language-specific instructions](/code_navigation/how-to/ configure_auto_indexing#private-repositories-and-packages-configuration)
+
+## Configure Code Navigation auto-indexing
+
+<Callout type="info">This feature is in beta for self-hosted customers.</Callout>
+
+Precise code navigation [auto-indexing](/code_navigation/explanations/auto_indexing) jobs are scheduled based on two fronts of configuration.
+
+The first front selects the set of repositories and commits within those repositories that are candidates for auto-indexing. These candidates are controlled by [configuring auto-indexing policies](#configure-auto-indexing-policies).
+
+The second front determines the set of index jobs that can run over candidate commits. By default, index jobs are [inferred](/code_navigation/explanations/auto_indexing_inference) from the repository structure's on disk. Index job inference uses heuristics such as the presence or contents of particular files to determine the paths and commands required to index a repository.
+
+Alternatively, index job configuration [can be supplied explicitly](#explicit-index-job-configuration) for a repository when the inference heuristics are not powerful enough to create an index job that produces the correct results. This might be necessary for projects that have non-standard or complex dependency resolution or pre-compilation steps, for example.
+
+### Configure auto-indexing policies
+
+Let's learn how to configure policies to control the scheduling of precise code navigation indexing jobs. Indexing jobs [produce a code graph data index](/code_navigation/explanations/precise_code_navigation) and uploads it to your Sourcegraph instance for use with code navigation.
```

Comment:
```suggestion
Let's learn how to configure policies to control the scheduling of precise code navigation indexing jobs. An indexing jobs [produces a code graph data index](/code_navigation/explanations/precise_code_navigation) and uploads it to your Sourcegraph instance for use with code navigation.
```

Since the verb in the second clause is 'uploads', having 'indexing jobs' earlier is incongruous.